ext/mbstring/mbstring.c

```diff
@@ -2949,28 +2949,38 @@ PHP_FUNCTION(mb_convert_kana)
 				DISALLOW_FLAG(MBFL_FILT_TL_ZEN2HAN_ALPHA, 'A', 'r');
 				DISALLOW_FLAG(MBFL_FILT_TL_ZEN2HAN_NUMERIC, 'A', 'n');
 				opt |= MBFL_FILT_TL_HAN2ZEN_ALL;
+				opt &= ~(MBFL_FILT_TL_HAN2ZEN_ALPHA | MBFL_FILT_TL_HAN2ZEN_NUMERIC); /* No need for these flags, they're covered */
```

Comment:
I think it might make more sense to instead make HAN2ZEN_ALL a union of HAN2ZEND_ALPHA etc. It would no longer be explicitly checked in the implementation.